src/meta/src/hummock/compaction_scheduler.rs

```diff
@@ -108,23 +108,23 @@ where
                     break 'compaction_trigger;
                 }
             };
-            self.pick_and_assign(compaction_group, request_channel.clone())
-                .await;
+            const SCHEDULE_COMPACTION_COUNT: usize = 4;
+            request_channel.unschedule(compaction_group);
+            for _ in 0..SCHEDULE_COMPACTION_COUNT {
+                if !self.pick_and_assign(compaction_group).await {
+                    break;
+                }
+            }
```

Comment:
Before this PR, we ensure that we will keep polling to see whether we have some compaction work to do.
After this PR, we only poll 4 times and stop polling untill there is a compaction request generated.

Will this be a problem?

Reply:
🤔 I would rollback it to main branch. I just change it because I suspend the compaction does not be scheduled after flush end.